### PR TITLE
Update BBEs

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
This PR will update `function_pointers`, `closures` and `anonymous_functions` BBEs

### CODE REVIEW

File: examples/anonymous-functions/anonymous_functions.description [hunk 1]
```diff
@@ -1,3 +1,3 @@
 // In Ballerina, anonymous functions are a syntactic shortcut for defining inline functions.
-// When you compare an anonymous function to a normal function definition in Ballerina,
+// When you compare an anonymous function to a normal function definition,
 // the only missing part is the function name.
```
Comment (MaryamZi) [inline examples/anonymous-functions/anonymous_functions.description]: Shall we say something like
```
// The only difference between an anonymous function and a normal function definition is that 
// an anonymous function does not have a name.
```

File: examples/anonymous-functions/anonymous_functions.bal [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+    io:println("Output: " + anonFunction2.call("Hello ", "World.!!!"));
+
     // If an anonymous function contains only the return statement in the body,
-    // you can use the arrow function expression instead of the anonymous function.
-    // You can infer the type of input parameters from the lhs.
-    function (string, string) returns (string) arrowExpression =
-                                                (x, y) => x + y;
-    io:println("Output: " + arrowExpression.call("Hello ", "World.!!!"));
+    // you can use the `arrow function expression` instead of the anonymous function.
+    // The types of the input parameters will be inferred from the lhs.
```
Comment (samgnaniah) [inline examples/anonymous-functions/anonymous_functions.bal]:
```suggestion
    // The types of the input parameters are inferred from the left-hand side.
```

File: examples/anonymous-functions/anonymous_functions.bal [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
+    // The return of the arrow function expression is determined by the
+    // evaluation of the expression to the right hand side of the `=>` symbol.
```
Comment (samgnaniah) [inline examples/anonymous-functions/anonymous_functions.bal]:
```suggestion
    // evaluation of the expression to the right-hand side of the `=>` symbol.
```

File: examples/anonymous-functions/anonymous_functions.description [hunk 4]
```diff
@@ -1,3 +1,3 @@
 // In Ballerina, anonymous functions are a syntactic shortcut for defining inline functions.
-// When you compare an anonymous function to a normal function definition in Ballerina,
+// When you compare an anonymous function to a normal function definition,
```
Comment (MaryamZi) [inline examples/anonymous-functions/anonymous_functions.description]:
```suggestion
// The only difference between an anonymous function and a normal function definition is that 
```

File: examples/anonymous-functions/anonymous_functions.description (hunk 1 above)
Comment (MaryamZi) [inline examples/anonymous-functions/anonymous_functions.description]:
```suggestion
// an anonymous function does not have a name.
```

**Pull request merged**